ModelBinders/GovUkMandatoryDateBinder.cs

```diff
@@ -0,0 +1,87 @@
+﻿using Microsoft.AspNetCore.Mvc.ModelBinding;
+using System.Linq;
+using System.Threading.Tasks;
+using GovUkDesignSystem.Attributes.DataBinding;
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
+
+namespace GovUkDesignSystem.ModelBinders
+{
+    /// <summary>
+    /// This model binder can be used to replace the default MVC model binder for a required date property. It will add
+    /// validation messages to the model state inline with the GovUk Design System guidelines.
+    /// This binder must be used alongside a GovUkDataBindingDateErrorTextAttribute attribute.
+    /// </summary>
+    public class GovUkMandatoryDateBinder : IModelBinder
+    {
+        public Task BindModelAsync(ModelBindingContext bindingContext)
+        {
+            var errorText = bindingContext.ModelMetadata.ValidatorMetadata.OfType<GovUkDataBindingDateErrorTextAttribute>().SingleOrDefault();
+            if (errorText == null)
+            {
+                throw new Exception("When using the GovUkMandatoryDateBinder you must also provide a GovUkDataBindingDateErrorTextAttribute attribute and ensure that you register GovUkDataBindingErrorTextProvider in your application's Startup.ConfigureServices method.");
+            }
+            var modelName = bindingContext.ModelName;
+            var modelSuffixes = new[] { "day", "month", "year" };
```

Comment:
As these values are shared between here and the HtmlGenerator I think it would be good to make an enum of the values and use that.

Reply:
using constants in html generator